Description: do not add content-length for status 204 response
Author: Hiroaki Nakamura <hnakamur@gmail.com>
Origin: other, https://github.com/apache/trafficserver/pull/9877
Last-Update: 2023-06-22
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/proxy/http/HttpTransact.cc
+++ b/proxy/http/HttpTransact.cc
@@ -6816,7 +6816,9 @@
         change_response_header_because_of_range_request(s, header);
         s->hdr_info.trust_response_cl = true;
       } else {
-        header->set_content_length(cl);
+        if (header->status_get() != HTTP_STATUS_NO_CONTENT) {
+          header->set_content_length(cl);
+        }
         s->hdr_info.trust_response_cl = true;
       }
     } else {
